### PR TITLE
clang: update to 5.0.0

### PR DESCRIPTION
**Pull request by mati865**
CC @martell

### CODE REVIEW

File: mingw-w64-clang/PKGBUILD [hunk 1]
```diff
@@ -153,58 +132,40 @@ prepare() {
...
-  patch -p1 -i "${srcdir}/0004-Globally-provide-DESTDIR-for-all-platforms.patch"
-  patch -p1 -i "${srcdir}/0005-not-install-msvc-files.patch"
-  patch -p1 -i "${srcdir}/0006-llvm-config-look-for-unversioned-shared-lib-on-win32.patch"
-  patch -p1 -i "${srcdir}/0007-add-pthread-as-system-lib-for-mingw.patch"
+  patch -p1 -i "${srcdir}/0001-mingw-w64-use-MSVC-style-ByteAlignment.patch"
+  patch -p1 -i "${srcdir}/0002-Fix-GetHostTriple-for-mingw-w64-in-msys.patch"
+  patch -p1 -i "${srcdir}/0003-Globally-provide-DESTDIR-for-all-platforms.patch"
+  patch -p1 -i "${srcdir}/0004-llvm-config-look-for-unversioned-shared-lib-on-win32.patch"
+  patch -p1 -i "${srcdir}/0005-add-pthread-as-system-lib-for-mingw.patch"
   patch -p1 -i "${srcdir}/0008-fix-bugpoint-multiple-definition.patch"
```
Comment (martell) [inline mingw-w64-clang/PKGBUILD]: maybe change this patch to from `0008` -> `0006`

**Comment by martell** [comment]
LGTM
We should start using `mingw-w64-clang-svn` because there are too many LLD patches to backport.

File: mingw-w64-clang/PKGBUILD (hunk 1 above)
Comment (mati865) [reply mingw-w64-clang/PKGBUILD]: Nice catch

**Comment by mati865** [comment]
lldb patch has been malformed, give me couple minutes.

**Comment by mati865** [comment]
Looks like branch `release_50` and tarball differs a lot for lldb.
I won't be able to spend another 5 hours to build it today so lets see how far CI can go.

**Comment by Alexpux** [comment]
is it ready for use?

**Comment by mati865** [comment]
I'm testing rebased patch and will upload it when build is finished.

**Comment by mati865** [comment]
I recreated patch from the ground using tarball.
It should build now but you can wait for CI.

**Comment by mati865** [comment]
Appveyor reached 74% before time limit.

**Pull request merged**